--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -137,12 +137,6 @@
 JAXLIB_VERSION = "0.9.2"
 
 python = use_extension("@rules_python//python/extensions:python.bzl", "python")
-python.override(
-    add_runtime_manifest_files = ["//third_party/py:rules_python_cpython_3_15.txt"],
-    minor_mapping = {
-        "3.15": "3.15.0rc1",
-    },
-)
 python.defaults(
     # The environment variable takes precedence if set.
     python_version = DEFAULT_PYTHON_VERSION,
@@ -154,7 +148,7 @@
 python.toolchain(python_version = "3.15")
 
 ### Pypi
-pip = use_extension("@rules_python//python/extensions:pip.bzl", "pip", dev_dependency = True)
+pip = use_extension("@rules_python//python/extensions:pip.bzl", "pip")
 
 [pip.parse(
     download_only = True,
@@ -163,18 +157,6 @@
         "numpy": ["numpy_headers"],
     },
     hub_name = "jax_pypi",
-    local_wheels = {
-        "jax": "dist/jax-*.whl",
-        "jaxlib": "dist/jaxlib-*.whl",
-        "jax-cuda12-pjrt": "dist/jax_cuda12_pjrt-*.whl",
-        "jax-cuda12-plugin": "dist/jax_cuda12_plugin-*.whl",
-        "jax-cuda13-pjrt": "dist/jax_cuda13_pjrt-*.whl",
-        "jax-cuda13-plugin": "dist/jax_cuda13_plugin-*.whl",
-        "libtpu": "dist/libtpu-*.whl",  # On CI, we use the downloaded nightly version of libtpu.
-        "ml_dtypes": "dist/ml_dtypes-*.whl",
-        "numpy": "dist/numpy-*.whl",  # On TSAN CI, we use a custom TSAN-instrumented NumPy wheel.
-        "scipy": "dist/scipy-*.whl",
-    },
     python_version = python_version,
     requirements_by_platform = {} if python_version in [
         "3.12",
